### PR TITLE
Mycelium UI

### PR DESCRIPTION
**Pull request by maximevanhees**
Related to #340

At this moment the Dioxus GUI offers the following:
- Changing API server address
- Viewing/searching connected peers
- Viewing rx and tx data flow of a specific peer
- Viewing/searching selected and fallback routes
- Removing peers
- Adding peers

Tested on Linux and Windows and MacOS.

Some screenshots:
Screenshot_20240813_142710
Screenshot_20240813_142826
Screenshot_20240813_142839

### CODE REVIEW

**Comment by maximevanhees** [comment]
In the meanwhile I've tested on MacOS and there seems to be an issue with rendering the CSS, causing the application to look horrible. Will update ASAP.

**Comment by LeeSmet** [comment]
@maximevanhees please rebase this branch on latest master since it's quite behind already. Additionally, in mycelium-ui/Cargo.toml, please fill in the metadata according to the other packages (author, version, license, etc).

**Comment by maximevanhees** [comment]
> @maximevanhees please rebase this branch on latest master since it's quite behind already. Additionally, in mycelium-ui/Cargo.toml, please fill in the metadata according to the other packages (author, version, license, etc).

Should be all set now.

File: mycelium-ui/src/main.rs [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+#[derive(Clone)]
+pub enum SortDirection {
+    Ascending,
+    Descending,
+}
+
+enum AppError {
+    NetworkError(reqwest::Error), // reqwest errors
+    AddressError(String),         // address parsing errors
+}
```
[the remark this answers is not in the document]
Comment (maximevanhees) [reply mycelium-ui/src/main.rs]: As this was only used once in the application, I removed all the code related to `AppError` completely and changed it to just use `reqwest::Error`.

File: mycelium-ui/src/components/mod.rs [hunk 2]
```diff
@@ -0,0 +1,4 @@
+pub mod home;
+pub mod layout;
+pub mod peers;
+pub mod routes;
```
[the remark this answers is not in the document]
Comment (maximevanhees) [reply mycelium-ui/src/components/mod.rs]: Fixed.

File: mycelium-ui/README.md [hunk 3]
```diff
@@ -0,0 +1,7 @@
+# Development
+
+Run the following command in the root of the project to start the Dioxus dev server:
+
+```bash
+dx serve --hot-reload --platform desktop
+```
```
[the remark this answers is not in the document]
Comment (maximevanhees) [reply mycelium-ui/README.md]: Updated.

**Comment by maximevanhees** [comment]
@LeeSmet
The bundling does indeed fail even though we still get a working binary. I have found this issue online, but there seem to be no apparent fix yet.

https://github.com/tauri-apps/tauri/issues/749
https://github.com/DioxusLabs/dioxus/issues/2344

**Comment by maximevanhees** [comment]
I have tested the bundling on Windows and there seems to be some error caused by an `Result::unwrap()` on a `Err` value from internal Dioxus code. It still managed to produce a working binary though. 

On MacOS the bundling works fine.

File: mycelium-ui/README.md [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+The myceliumd daemon is the background process that manages the Mycelium network connection 
+and provides the data that the dashboard application displays. For more information on setting up and
+running `myceliumd`, please read [this](../README.md).
+
+Once you have the Dioxus CLI installed, you can build and run the application in development mode using
+the following command (in the `mycelium-ui` directory):
+
+`dx serve`
+
+This will start a development server and launch the application in your default webbrowser.
```
Comment (LeeSmet) [inline mycelium-ui/README.md:23]: Is it really your default webbrowser? looks more like a webview

**Review by LeeSmet — CHANGES_REQUESTED** [review]
This PR looks good to go, except for some minor documentation nitpicks

**Review by LeeSmet — APPROVED** [review]
LGTM :tada:

**Pull request merged**